### ISSUE TITLE
initialSelection takes a storyId string not object

### ISSUE THREAD

**Issue by henrymoulton** [storybookjs/react-native#161]
**Describe the bug**

> initialSelection: Object (null)
>        -- initialize storybook with a specific story. In case a valid object is passed, it will take precedence over `shouldPersistSelection. ex: `{ kind: 'Knobs', story: 'with knobs' }`

`initialSelection` actually takes a storyId string because `checkStory` takes a storyId string

https://github.com/storybookjs/react-native/blob/1fd61890534dee184b1aaffd169dd83211ef487a/app/react-native/src/preview/index.tsx#L187

```
checkStory(storyId: string) {
    if (!storyId) {
      return null;
    }
    const story = this._getStory(storyId);
    if (story.storyFn === null) {
      return null;
    }
    return story;
  }
```

**Comment by henrymoulton** [comment]
Follow up questions:


1. What is the function that generates storyIds? It's clearly generated based off a Kind + Story but because Storybook is split across lots of small packages it's hard to follow.

I've written an initial guess:

```
const convertToStorybooksKebabCase = (string: string) => {
  let result = string;
  result = result.replace(/\s+/g, '-');

  // Looks like the Storybook IDs also contain no punctuation.
  // TODO: Complete this by finding where they create/format the ID in the codebase.
  result = result.replace(/\(/g, '');
  result = result.replace(/\)/g, '');
  return result.toLowerCase();
};
```

2. This relates to my other question in https://github.com/storybookjs/react-native/issues/154, how can we access all storyIDs?

**Comment by dannyhw** [comment]
it seems like it would make sense for the initial selection to take an object, it's strange that it doesn't. 

If you would like to open a PR to fix this I'd happily accept it. Otherwise I will add this to my backlog and change it when I can.

Thanks for reporting this :)

**Comment by scousino** [comment]
While waiting, can we get some kind of official format for the storyIds? Otherwise this prop is unusable

**Comment by scousino** [comment]
For anyone interested you can import the toId function an use it to create the story id.
`import { toId } from '@storybook/csf';`

However, even after doing this and passing the ID to initialSelection, nothing happens on load. I still get the default Preview screen

**Comment by dannyhw** [comment]
See the toId function here.

https://github.com/ComponentDriven/csf/blob/next/src/index.ts


For storiesof("foo").add("bar")

That should produce something like 
foo--bar all lowercase and most special characters are replaced with -. 

I know there isn't really a nice way to get at this ID right now and I plan on improving that. Apologies for not following up on this until now.

**Comment by dannyhw** [comment]
> For anyone interested you can import the toId function an use it to create the story id.
> `import { toId } from '@storybook/csf';`
> 
> However, even after doing this and passing the ID to initialSelection, nothing happens on load. I still get the default Preview screen

Hmm I'll have to check that, could be a bug. Thanks for reporting that.

**Comment by scousino** [comment]
> 
> 
> > For anyone interested you can import the toId function an use it to create the story id.
> > `import { toId } from '@storybook/csf';`
> > However, even after doing this and passing the ID to initialSelection, nothing happens on load. I still get the default Preview screen
> 
> Hmm I'll have to check that, could be a bug. Thanks for reporting that.

I created a new issue for that specific bug (#273), since its separate from the string vs. object issue of initialSelection docs. My 2 cents, I think it makes more sense for initialSelection to take an object and then create the ID itself. Rather than consumers having to create the ID

**Comment by dannyhw** [comment]
> > > For anyone interested you can import the toId function an use it to create the story id.
> > > `import { toId } from '@storybook/csf';`
> > > However, even after doing this and passing the ID to initialSelection, nothing happens on load. I still get the default Preview screen
> > 
> > 
> > Hmm I'll have to check that, could be a bug. Thanks for reporting that.
> 
> I created a new issue for that specific bug (#273), since its separate from the string vs. object issue of initialSelection docs. My 2 cents, I think it makes more sense for initialSelection to take an object and then create the ID itself. Rather than consumers having to create the ID

@scousino thanks for creating that issue I'll add it to the backlog

**Comment by dannyhw** [comment]
in #274 I've made an initial attempt to fix this. I might change the configuration of this to the main.js file later though.
